Add an error code for when the path fails (#4331)
quicwg/base-drafts: draft-ietf-quic-transport.md

unusable.  This does not necessarily imply a failure of the connection -
endpoints can continue sending packets over other paths as appropriate.  If no
paths are available, an endpoint can wait for a new path to become available or
close the connection.  An endpoint that has no valid network path to its peer
MAY signal this using the NO_VIABLE_PATH connection error, noting that this is

> only possible if the network has a limited MTU.

Reviewer: ```suggestion
only possible if the network path exists but does not support the required
MTU {{datagram-size}}.
```

: An endpoint has reached the confidentiality or integrity limit for the AEAD
  algorithm used by the given connection.

NO_VIABLE_PATH (0x10):

: An endpoint has determined that the network path is incapable of supporting
  QUIC.  An endpoint is unlikely to receive CONNECTION_CLOSE carrying this code

>   unless the reason is to do the path not supporting a large enough MTU.

Reviewer 2: ```suggestion
  except when the path does not support a large enough MTU.
```
